### DISCUSSION TITLE
Bars: design, configuration, and placement

### DISCUSSION THREAD

**Discussion by Simplenuity** [zim-desktop-wiki/zim-desktop-wiki#1612] [General]
Following up on issues like [#1595](https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1595) and [#1594](https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1594) this is intended to discuss the bars, especially with the changes that came along with version 0.74

Available bars:
* edit bar
* menu bar
* attachment bar (plugin)
* bookmark bar (plugin)
* path bar (plugin)
* tool bar (plugin)

Questions (this list is not exhaustive, rather a starting point for the discussion):

* What should be shown by default on which bar?

* Could they be made customizable (which would cover different needs and avoid lots of discussions)?

* Which one could be made optional?

* Where to place them, which options could be made available? Could the bottom of a side pane also be used, for example for (the information of) the old status bar?

Quite a few questions would need Jaap's input regarding feasibility and his further roadmap.

**Comment by jaap-karssenberg** [comment]
Attached my sketch of where I plan to go with the header bar and how it
could look without client side window decoration.

Op di 21 sep. 2021 23:02 schreef Simplenuity ***@***.***>:

**Reply by jaap-karssenberg** [reply]
Now with attachment ... discussion forum apparently does not allow attachment via email
![IMG_20210922_102219](https://user-images.githubusercontent.com/1248286/134309774-9b55fbf8-5001-4b73-8736-06739aa8e683.jpg)

**Reply by paapereira** [reply]
I'm OK with the classic look.

**Reply by introt** [reply]
Are you planning to add a right-click history menu to navigation, similar to what the web browsers have? In addition, the home button could expose bookmarks saved using the bookmarks plugin, or list the pages linked from home.

**Comment by zinnschlag** [comment]
This at least addresses the problem of Zim looking completely alien on a non-Gnome 3 desktop. It still uses additional screen space and adds a lot of visual noise (compared to classic with menu bar and no other bars). One of the main advantages of the old UI was its flexibility. Individual elements could be configured on/off and so it supported multiple styles of working; including a truly minimalist one for people who never use toolbars in primarily keyboard-driven applications.

I am not sure how well this design would function for people with different work styles, but as a minimalist I would need following feature: All functions of the toolbar should be reachable via keyboard-shortcut (including the hamburger menu) and it should be possible to configure the toolbar away. An easy way to bring the toolbar back (and therefore not lock the user out of the UI when he forgets a keyboard-shortcut) would be an item in the context menu. The default terminal under the Mate desktop environment does that for example.

This leaves the issue of discoverability. The old menu bar was very useful in this regard: A single tree structure that contains all main functions. Much easier to find something compared to an UI that scatters functionality across several elements (toolbar, hamburger menu), of which some don't even have any structure and are presented as a flat list instead. Unfortunately I have no idea how to address this problem within the given design.

**Reply by blueclochard** [reply]
This is also how I use Zim: with just a classic menu and no toolbars. It is very efficient, not in the least because it works the same in all other programs. With a hamburger menu, it is up to the programmer where functions are placed. Unfortunately, there doesn't seem to be a consensus on which functions should be placed where under the hamburger if at all. Also, as I mentioned in #1582, a hamburger menu adds at least one and for some actions likely more steps to achieve the same.

Apologies for replying in #1582 by the way. I now see my reply there would have been better off here in the Discussions section. I read somewhere Jaap has his hands in his hair already because of the hundreds of open issues (at least he has a lot of hair).

Incidentally, with at least a classic, non-CSD title bar, the issues Windows users have such as #1639 would probably be solved as well.

**Comment by jaap-karssenberg** [comment]
I hear you. Had not considered the use case of no window decoration *and*
no toolbar. Sounds like there should be a menubar option after all. Will
draw a new proposal.

**Reply by arkilho** [reply]
Thank you for considering this option Jaap.
I'll wait for that feature before upgrading.
And thank you also for keeping the file data set consistent across upgrades and also downgrades, as I just experienced..

**Reply by timelyfiasco** [reply]
Just want to share my screen, FWIW. 
![image](https://user-images.githubusercontent.com/33004592/136680748-303d0878-56fa-4adf-8798-9aa1c748626f.png)
The default view looks better after I got rid of the edit bar by accident.

**Comment by Simplenuity** [comment]
As I suppose it's rather a feature than a bug, I'll bring it up here (please feel free to move over to issues if appropriate)

**Describe the "bug"**
When I activate "Show edit bar along bottom of editor" (Preferences > Interface) WHILE also have active the plugin Toolbar with "Include formatting tools in toolbar" and "Position in the window: Top" I will see the edit bar along bottom of editor. But as soon as I open another notebook the option gets deactivated and the edit bar along bottom of editor will disappear in all opened notebooks.

The main reason I (try to) activate this option is to see the status bar, specifically the level of heading

**To Reproduce**
Steps to reproduce the behavior:
1. Have plugin Toolbar with "Include formatting tools in toolbar" and "Position in the window: Top" active
2. Activate under Preferences > Interface the option "Show edit bar along bottom of editor" and click OK (which will show the edit bar at the bottom too.
3. Open a second notebook or close/open the current one.
4. edit bar along bottom of editor has disappeared, option has been deactivated

**Expected behavior**
to have edit bar along bottom of editor permanently active (despite being integrated in Toolbar on top too)

**Environment (please complete the following information):**
 - Version: 0.74.2
 - OS: Linux Lite 4.8 (based on Ubuntu 18.04)
 - Language: English

As mentioned above, main reason for trying to activate it is to get back the status bar.
So as solution it seems more reasonable to have the info of the status bar shown somewhere else than having two edit bars active...
